cnxepub/formatters.py

```diff
@@ -317,15 +319,36 @@ def _fix_namespaces(html):
     return etree.tostring(let, pretty_print=True, encoding='utf-8')
 
 
-def exercise_callback_factory(match, url_template):
+def _replace_tex_math(node, mml_url):
+    """call mml-api service to replace TeX math in body of node with mathml"""
+
+    res = requests.post(mml_url, {'math': node.text,
+                                  'mathType': 'TeX',
+                                  'mml': 'true'})
+    if res:
+        eq = res.json()
+        mml = etree.fromstring(eq['components'][0]['source'])
+        if node.tag.endswith('span'):
+            mml.set('display', 'inline')
+        elif node.tag.endswith('div'):
+            mml.set('display', 'block')
+        return mml
+    else:
+        return None
```

Comment:
If the code should never be executed should this line throw an error to make it clear that this code path is never evaluated?